core/src/main/scala/org/genivi/sota/core/VehicleUpdatesResource.scala

```diff
@@ -183,7 +183,9 @@ class VehicleUpdatesResource(db : Database, resolverClient: ExternalResolverClie
       put {
         path("installed") { updateInstalledPackages(vin) } ~
         path("order") { setInstallOrder(vin) } ~
-        (extractUuid & path("cancelupdate") ) { uuid => cancelUpdate(vin, uuid) }
+        (extractUuid & path("cancelupdate") ) { uuid =>
+          namespaceExtractor { ns => cancelUpdate(ns, vin, uuid) }
```

Comment:
Concretely, the workaround (using `Option`) looks as follows:

```diff
--- a/core/src/main/scala/org/genivi/sota/core/VehicleUpdatesResource.scala
+++ b/core/src/main/scala/org/genivi/sota/core/VehicleUpdatesResource.scala
@@ -161,9 +161,9 @@ class VehicleUpdatesResource(db : Database, resolverClient: ExternalResolverClie
   /**
     * The web app PUT the status of the given ([[UpdateSpec]], VIN) to [[UpdateStatus.Canceled]]
     */
-  def cancelUpdate(ns: Namespace, vin: Vehicle.Vin, uuid: Refined[String, Uuid]): Route = {
+  def cancelUpdate(nsOpt: Option[Namespace], vin: Vehicle.Vin, uuid: Refined[String, Uuid]): Route = {
     val response =
-      db.run(UpdateSpecs.cancelUpdate(ns, vin, uuid))
+      db.run(UpdateSpecs.cancelUpdate(nsOpt, vin, uuid))
           .map(_ => StatusCodes.NoContent)
           .recover { case _ => StatusCodes.BadRequest }
           .map(HttpResponse(_))
@@ -184,7 +184,7 @@ class VehicleUpdatesResource(db : Database, resolverClient: ExternalResolverClie
         path("installed") { updateInstalledPackages(vin) } ~
         path("order") { setInstallOrder(vin) } ~
         (extractUuid & path("cancelupdate") ) { uuid =>
-          namespaceExtractor { ns => cancelUpdate(ns, vin, uuid) }
+          namespaceExtractor { ns => cancelUpdate(Some(ns), vin, uuid) }
         }
       } ~
       post {
diff --git a/core/src/main/scala/org/genivi/sota/core/db/UpdateSpecs.scala b/core/src/main/scala/org/genivi/sota/core/db/UpdateSpecs.scala
index b17843e..51fddac 100644
--- a/core/src/main/scala/org/genivi/sota/core/db/UpdateSpecs.scala
+++ b/core/src/main/scala/org/genivi/sota/core/db/UpdateSpecs.scala
@@ -123,6 +123,15 @@ object UpdateSpecs {
   }
 
   /**
+    * TODO use instead the overloaded version of this method taking a namespace.
+    */
+  private def queryBy(vin: Vehicle.Vin,
+                      requestId: UUID): Query[UpdateSpecTable, UpdateSpecTableRowType, Seq] = {
+    updateSpecs
+      .filter(row => row.vin === vin && row.requestId === requestId)
+  }
+
+  /**
     * Lookup by PK in [[UpdateSpecTable]] table.
     * Note: A tuple is returned instead of an [[UpdateSpec]] instance because
     * the later would require joining [[RequiredPackageTable]] to populate the `dependencies` of that instance.
@@ -218,10 +227,14 @@ object UpdateSpecs {
     *
     * @param uuid of the [[UpdateRequest]] being cancelled for the given VIN
     */
-  def cancelUpdate(ns: Namespace, vin: Vehicle.Vin, uuid: Refined[String, Uuid])
+  def cancelUpdate(nsOpt: Option[Namespace], vin: Vehicle.Vin, uuid: Refined[String, Uuid])
                   (implicit executor: ExecutionContext): DBIO[Unit] = {
-    queryBy(ns, vin, UUID.fromString(uuid.get))
-      .filter(_.status === UpdateStatus.Pending)
+    val q = nsOpt match {
+      case None     => queryBy(    vin, UUID.fromString(uuid.get))
+      case Some(ns) => queryBy(ns, vin, UUID.fromString(uuid.get))
+    }
+
+     q.filter(_.status === UpdateStatus.Pending)
       .map(_.status)
       .update(UpdateStatus.Canceled)
       .flatMap { rowsAffected =>
```